src/gui/GuiApplication.cpp

```diff
@@ -240,6 +244,19 @@ void GuiApplication::childDestroyed(QObject *obj)
 	}
 }
 
+void GuiApplication::sendInterrupt(int signal) {
+	emit interruptOccurred(signal);
+}
+
+// Slot for handling interrupts from main()
+void GuiApplication::processInterrupt(int signal) {
+	switch(signal) {
+		case SIGINT:
+			// add last minute resource clean-up goes here
+			qApp->exit(3);
```

Comment:
`exit()` seems abrupt, but if we call `quit()`, `MainApplication` will start asking questions which no one wants.  I checked the clean-up code and it primarily deletes a recovery file, which we can't possibly know if should or should not be done, so I believe this will run any necessary destructors from a memory perspective and be as clean as possible.  If this changes down the road, I would advise we move this slot into `MainApplication`, but I kept it here for simplicity.